### ISSUE TITLE
Segmentation fault with texture modifiers

### ISSUE THREAD

**Issue by a-tour-ist** [minetest/minetest#14242]
### Minetest version

```true
a7eaee77ca0800553589b8ed716a69451496970b
```

### Active renderer

OpenGL 4.6

### Irrlicht device

X11

### Operating system and version

Manjaro Linux

### CPU model

Intel© Core™ i7-7600U CPU @ 2.80GHz × 2

### Summary

When setting width or hight in texture modifiers to huge values (2^32 - 1 or -1 bc. they are casted to u32) the client crashes with a Segmentation fault.

I was able to reproduce with:
```
[combine
[resize
[fill
```

example stacktrace from [combine:
```
Thread 1 "minetest" received signal SIGSEGV, Segmentation fault.
0x0000555555ec58ff in irr::video::CImage::getPixel (this=0x55555ed30c70, x=0, y=1) at /home/tour/minetest/lib/irrlichtmt/source/Irrlicht/CImage.cpp:105
105			return ((u32*)Data)[y*Size.Width + x];
#0  0x0000555555ec58ff in irr::video::CImage::getPixel (this=0x55555ed30c70, x=0, y=1) at /home/tour/minetest/lib/irrlichtmt/source/Irrlicht/CImage.cpp:105
#1  0x000055555590a3cf in blit_with_alpha (src=0x55555ed30e70, dst=0x55555ed30c70, src_pos=..., dst_pos=..., size=...) at /home/tour/minetest/src/client/tile.cpp:2135
#2  0x00005555559046ab in TextureSource::generateImagePart (this=0x5555569b73b0, part_of_name="[combine:-1x-1:0,0=default_cobble.png:0,16=default_wood.png", baseimg=@0x7fffffffbc10: 0x55555ed30c70, source_image_names=std::set with 1 element = {...}) at /home/tour/minetest/src/client/tile.cpp:1344
#3  0x000055555590311b in TextureSource::generateImage (this=0x5555569b73b0, name="[combine:-1x-1:0,0=default_cobble.png:0,16=default_wood.png", source_image_names=std::set with 1 element = {...}) at /home/tour/minetest/src/client/tile.cpp:1066
#4  0x000055555590083c in TextureSource::generateTexture (this=0x5555569b73b0, name="[combine:-1x-1:0,0=default_cobble.png:0,16=default_wood.png") at /home/tour/minetest/src/client/tile.cpp:644
#5  0x00005555559003ae in TextureSource::getTextureId (this=0x5555569b73b0, name="[combine:-1x-1:0,0=default_cobble.png:0,16=default_wood.png") at /home/tour/minetest/src/client/tile.cpp:508
#6  0x0000555555900d74 in TextureSource::getTexture (this=0x5555569b73b0, name="[combine:-1x-1:0,0=default_cobble.png:0,16=default_wood.png", id=0x0) at /home/tour/minetest/src/client/tile.cpp:697
#7  0x000055555598eb3b in GUIFormSpecMenu::parseImage (this=0x55555f26afa0, data=0x7fffffffc670, element="1,1;5,5;[combine:-1x-1:0,0=default_cobble.png:0,16=default_wood.png") at /home/tour/minetest/src/gui/guiFormSpecMenu.cpp:790
#8  0x00005555559a215b in GUIFormSpecMenu::parseElement (this=0x55555f26afa0, data=0x7fffffffc670, element="\nimage[1,1;5,5;[combine:-1x-1:0,0=default_cobble.png:0,16=default_wood.png") at /home/tour/minetest/src/gui/guiFormSpecMenu.cpp:2888
#9  0x00005555559a4a08 in GUIFormSpecMenu::regenerateGui (this=0x55555f26afa0, screensize=...) at /home/tour/minetest/src/gui/guiFormSpecMenu.cpp:3392
#10 0x00005555559feff4 in GUIModalMenu::draw (this=0x55555f26afa0) at /home/tour/minetest/src/gui/modalMenu.cpp:82
#11 0x000055555596ae4c in irr::gui::IGUIElement::draw (this=0x55555647d6d0) at /home/tour/minetest/lib/irrlichtmt/include/IGUIElement.h:337
#12 0x0000555555e70351 in irr::gui::CGUIStaticText::draw (this=0x55555647d6d0) at /home/tour/minetest/lib/irrlichtmt/source/Irrlicht/CGUIStaticText.cpp:141
#13 0x000055555596ae4c in irr::gui::IGUIElement::draw (this=0x55555647a708) at /home/tour/minetest/lib/irrlichtmt/include/IGUIElement.h:337
#14 0x0000555555e50278 in irr::gui::CGUIEnvironment::drawAll (this=0x55555647a700, useScreenSize=true) at /home/tour/minetest/lib/irrlichtmt/source/Irrlicht/CGUIEnvironment.cpp:188
#15 0x000055555589e984 in DrawHUD::run (this=0x55555ed19c40, context=...) at /home/tour/minetest/src/client/render/plain.cpp:71
#16 0x000055555589ac2a in RenderPipeline::run (this=0x55555ebf89c0, context=...) at /home/tour/minetest/src/client/render/pipeline.cpp:286
#17 0x00005555558953ec in RenderingCore::draw (this=0x55555ed19cb0, _skycolor=..., _show_hud=true, _show_minimap=false, _draw_wield_tool=true, _draw_crosshair=true) at /home/tour/minetest/src/client/render/core.cpp:52
#18 0x00005555558a76ed in RenderingEngine::draw_scene (this=0x5555564abfe0, skycolor=..., show_hud=true, show_minimap=false, draw_wield_tool=true, draw_crosshair=true) at /home/tour/minetest/src/client/renderingengine.cpp:323
#19 0x000055555580d6a0 in Game::drawScene (this=0x7fffffffd2c0, graph=0x7fffffffd0c0, stats=0x7fffffffd080) at /home/tour/minetest/src/client/game.cpp:4325
#20 0x000055555580cb60 in Game::updateFrame (this=0x7fffffffd2c0, graph=0x7fffffffd0c0, stats=0x7fffffffd080, dtime=0.0167650003, cam=...) at /home/tour/minetest/src/client/game.cpp:4200
#21 0x00005555557fb4d4 in Game::run (this=0x7fffffffd2c0) at /home/tour/minetest/src/client/game.cpp:1315
#22 0x00005555558103e0 in the_game (kill=0x5555563f26c0 <porting::g_killed>, input=0x5555569f0d90, rendering_engine=0x5555564abfe0, start_data=..., error_message="", chat_backend=..., reconnect_requested=0x7fffffffd5d4) at /home/tour/minetest/src/client/game.cpp:4597
#23 0x00005555557915cc in ClientLauncher::run (this=0x7fffffffda40, start_data=..., cmd_args=...) at /home/tour/minetest/src/client/clientlauncher.cpp:264
#24 0x0000555555d1cb04 in main (argc=2, argv=0x7fffffffdee8) at /home/tour/minetest/src/main.cpp:266
```

### Steps to reproduce

view this formspec (breaking the examples from the lua api):
```
size[5,5]
image[0,0;1,1;[combine:-1x-1:0,0=default_cobble.png:0,16=default_wood.png]
image[0,1;1,1;default_sandstone.png^[resize:4294967295x4294967295]
image[0,0;1,1;[fill:-1x-1:red#f]
```

**Comment by appgurueu** [comment]
Yes, texture modifiers indeed lack much validation of the parameters. There are quite a few bugs like this. Please don't file separate issues for all of them - collect them all in one issue, and fix them in one PR (if you would like to fix them).

**Comment by a-tour-ist** [comment]
I guess this was solved by #14308 (bad dimension values) and #14314 (no longer attempting to read from pixels which are not part of the image)

I can't reproduce with the latest version anymore.
